.circleci/config.yml

```diff
@@ -1,72 +1,155 @@
-version: 2
+version: 2.1
 
-jobs:
-  build:
+executors:
+  node-latest:
+    working_directory: ~/repo
     docker:
       - image: circleci/node:latest
+
+jobs:
+  test-gen:
+    executor: node-latest
     steps:
       - checkout
       - run:
-          name: Update npm
-          command: 'sudo npm install -g npm@latest'
-      - run:
-          name: Get Rust
+          name: get rust
           command: "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y"
       - run:
-          name: Setup Env
+          name: setup env
           command: echo "export PATH=$HOME/.cargo/bin:$PATH" >> $BASH_ENV
+      - run:
+          name: update rust dependencies if releasing
+          command: |
+            if [[ "$CIRCLE_BRANCH" =~ release/v.* ]]; then
+              cargo install cargo-edit
+              grep branch Cargo.toml | grep develop | awk '{print $1}' | xargs -I[] cargo add --allow-prerelease []
+              sed -i 's/branch = "develop", //g' Cargo.toml
+              export CARGO_VER=$(echo $CIRCLE_BRANCH | sed 's:release/v::')
+              sed -i "s/^version.*/version = \"$CARGO_VER\"/" Cargo.toml
+            fi
+      - run:
+          name: update js dependencies if releasing
+          command: |
+            if [[ "$CIRCLE_BRANCH" =~ release/v.* ]]; then
+              sudo npm install -g kl-ci-tools
+              kl-ci-tools deps | xargs -I[] npm install []@latest
+              echo $CIRCLE_BRANCH | sed 's:release/::' | xargs npm --no-git-tag-version version
+            fi
       - restore_cache:
           key: dependency-cache-{{ checksum "package.json" }}
```

Comment:
Looks like this key needs updated
`git-sha-{{ .Revision }}{{ .Branch }}`

Reply:
Actually I think it can just be removed

Reply:
Or actually this wouldn't work would it?  Should the save cache save with the package.json still?